cmd/sts-errors.go

```diff
@@ -18,15 +18,23 @@ package cmd
 
 import (
 	"encoding/xml"
+	"fmt"
 	"net/http"
 
 	xhttp "github.com/minio/minio/cmd/http"
 )
 
 // writeSTSErrorRespone writes error headers
-func writeSTSErrorResponse(w http.ResponseWriter, err STSError) {
+func writeSTSErrorResponse(w http.ResponseWriter, errCode STSErrorCode, errCtxt error) {
+	err := stsErrCodes.ToSTSErr(errCode)
 	// Generate error response.
-	stsErrorResponse := getSTSErrorResponse(err, w.Header().Get(xhttp.AmzRequestID))
+	stsErrorResponse := STSErrorResponse{}
+	stsErrorResponse.Error.Code = err.Code
+	stsErrorResponse.RequestID = w.Header().Get(xhttp.AmzRequestID)
+	stsErrorResponse.Error.Message = err.Description
+	if errCtxt != nil {
+		stsErrorResponse.Error.Message = fmt.Sprintf("%s\n%v", err.Description, errCtxt)
```

Comment:
I would suggest we give errCtxt the right priority, since the Err.Code is correctly set we don't need to provide `\n` based multiple messages. It will confuse the XML parsers and harder. Following code is enough.

```suggestion
		stsErrorResponse.Error.Message = fmt.Sprintf("%v", errCtxt)
```